autocertkit/testbase.py

```diff
@@ -109,15 +112,32 @@ def run(self, debug=False, test_name=None):
 
             rec = {}
             try:
-                log.debug("******** %s.%s ********" %
-                          (self.__class__.__name__, test))
-                res = getattr(self, test)(self.session)
+                log.debug("******** %s.%s ********" % (self.__class__.__name__, test))
 
-                # If test executed without failure it can be either skipped or
-                # passed.
-                if 'skipped' in res and res['skipped']:
-                    rec['result'] = 'skip'
+                res = getattr(self, test)(self.session)
+                """
+                Critical key and value in res:
+                    'status': 'init'    initial status before running
+                              'running' test still running
+                              'done',   test finished
+                    'result': 'skip'    needless to run
+                              'pass'    test OK
+                              'fail'    test failed (with Exception occurs)
+                    'control': any private data of test itself.
+                    'superior': return common info to test runner from test, 
```

Comment:
Sorry I am not aware of my reply disappeared, but I think you should already received github notification mail.
If not please let me know, I will re-add.

Reply:
See my another reply
You can learn from SR-IOV test and Crashdump test

Reply:
So I think you need different names as `control` and `superior` don't make sense to me. 
Could `control` be `test_data`?
Could `superior` be `transient_data`? (given the fact the data is not persistent?).

Reply:
There is already another field in xml using 'test_data'
'transient_data' seems better, while it does not reveal the data is to just test runner
I suggest to just keep them, since no good name for reader to understand immediately, and reader would get known from comments and sample code.